### PR TITLE
locale - let it check interactive-sections again

### PR DESCRIPTION
**Pull request by dbungert**
LP: #1919251

### CODE REVIEW

**Comment by dbungert** [comment]
When this was introduced, the comment was:
  show the initial screen if any screens are interactive
  the idea here is to stop multiple subiquitys from trampling on each other

Is that still relevant?

**Comment by mwhudson** [comment]
Oh hah I'd forgotten that rationale. No, that's not a valid reason any more -- but there is still one thing I'd like to fix, which is to move the question we ask on the serial console about switching to 'rich mode' somewhere more generic so that it still gets asked even if locale is not interactive. Basically I think this means making a "Stretchy" to ask the question and showing it with add_global_overlay() just after starting the urwid event loop -- if you look at how error reports or cloud-init failures are reported at startup, like that!

**Comment by mwhudson** [comment]
Very nice, this is exactly what I asked for. Unfortunately (and this is my fault), this does have one regression in functionality though: we probably shouldn't offer a choice of languages in non-rich mode (as most non-English languages will just display as a mess of '?'s). So perhaps instead of being an overlay, the question about rich mode should be its own screen (complete with client/controllers/richmode I guess) that can skip on non-serial. Or we can add some way to skip past the language screen if it's showing when the user selects "continue in basic mode".

What do you think? Sorry to mess you around like this.

**Review by mwhudson — CHANGES_REQUESTED** [review]
This looks really good thanks! Still have a few comments but nothing deep.

File: subiquity/client/client.py [hunk 1]
```diff
@@ -419,6 +420,9 @@ def select_initial_screen(self):
             for i, controller in enumerate(self.controllers.instances):
                 if controller.name == last_screen:
                     index = i
+        # only show the first (Serial) screen if relevant
+        if index < 1 and not self.opts.run_on_serial:
+            index = 1
```
[the remark this answers is not in the document]
Comment (dbungert) [reply subiquity/client/client.py:425]: Ah, I like that better, good suggestion

File: console_conf/controllers/welcome.py [hunk 2]
```diff
@@ -29,8 +29,10 @@ def done(self):
         self.app.next_screen()
 
     def cancel(self):
-        # Can't go back from here!
-        pass
+        if not self.app.opts.run_on_serial:
```
[the remark this answers is not in the document]
Comment (dbungert) [reply console_conf/controllers/welcome.py:32]: Actually, I just noticed this is in console_conf, so I'm reverting this hunk.

**Comment by dbungert** [comment]
@mwhudson - thanks, all feedback items should be addressed, would you confirm?

**Review by mwhudson — APPROVED** [review]
Thanks, this is great. I guess you might want to rebase or squash merge or something, the history is a bit convoluted. If you can be bothered :)

**Pull request merged**